### ISSUE TITLE
Failed to handle key=<Esc>. Illegal argument: line must be non-negative

### ISSUE THREAD

**Issue by hosseinzamaninasab** [VSCodeVim/Vim#5226]
**To Reproduce**

1.  Go to '...'
2.  Click on '....'
3.  Scroll down to '....'
4.  See an error

**VSCodeVim version**: 1.17.0

<details><summary>Stack trace</summary>

```
Error: Failed to handle key=<Esc>. Illegal argument: line must be non-negative
    at Object.t.illegalArgument (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:10:662)
    at new b (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:499:364)
    at new b (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:497:187)
    at new l (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:1:9198)
    at h.updateAndReturnMarks (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:510380)
    at h.finishCurrentStep (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:513901)
    at F.runAction (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:477803)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
    at F.handleKeyAsAnAction (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:474707)
    at F.handleKeyEvent (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:471516)
    at Object.promise (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:231560)
    at Object.runTasks (c:\Users\Hossein\.vscode\extensions\vscodevim.vim-1.17.0\out\extension.js:9:621827)
```

</details>

**Comment by J-Fields** [comment]
What were you trying to do when this happened?

**Comment by hosseinzamaninasab** [comment]
Just wanna to scroll down with j instead of arrow keys

**Comment by berknam** [comment]
I think this is related to this part of code (line 363) but I'm not sure:
https://github.com/VSCodeVim/Vim/blob/8894fa6a608c928455dcb2fd6364b3605aa02ef9/src/history/historyTracker.ts#L360-L372

Line 381 below then uses a check to make sure it doesn't try to use negative lines, maybe line 363 should do the same?
https://github.com/VSCodeVim/Vim/blob/8894fa6a608c928455dcb2fd6364b3605aa02ef9/src/history/historyTracker.ts#L381

I have not been able to reproduce this myself but from the stack trace this seems to be the only place where that error could happen.

**Comment by J-Fields** [comment]
Thanks @berknam - I haven't reproduced this either but I agree that looks like the problem line. I've added bounds-checking as you suggested. Seems like a band-aid on an underlying issue I haven't yet identified, but better than nothing.

**Comment by J-Fields** [comment]
I think this was fixed by 081edbaf102b086befb078ae12b5ade3ce8a6ca8. Closing for now.
